.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,51 @@
+name: "scicomp-provisioner-gha"
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - "*"
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - uses: pre-commit/action@v3.0.0
+  sceptre-validate:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - name: 'Sceptre Validate dev'
+        uses: Sceptre/github-ci-action@master
+        with:
+          sceptre_version: '4.0.2'
+          sceptre_subcommand: 'validate dev'
+      - name: 'Sceptre Validate prod'
+        uses: Sceptre/github-ci-action@master
+        with:
+          sceptre_version: '4.0.2'
+          sceptre_subcommand: 'validate prod'
+  sceptre-deploy:
+    runs-on: ubuntu-latest
+    if: ${{ github.ref == 'refs/heads/master' }}
+    needs: [ 'pre-commit', 'sceptre-validate' ]
+    steps:
+      - name: 'Checkout'
+        uses: actions/checkout@master
+      - name: Assume AWS Role
+        uses: aws-actions/configure-aws-credentials@v1
```

Comment:
it's v2 now https://github.com/marketplace/actions/configure-aws-credentials-for-github-actions